zeppelin-web/src/app/handsontable/handsonHelper.js

```diff
@@ -0,0 +1,199 @@
+        }
+      });
+      if (TH.firstChild.lastChild.nodeName === 'BUTTON') {
+        TH.firstChild.removeChild(TH.firstChild.lastChild);
+      }
+      TH.firstChild.appendChild(button);
+      TH.style['white-space'] = 'normal';
+    }
+  };
+};
+
+/*
+** Private Service Functions
+*/
+
+function _addButtonMenuEvent(button, menu) {
+  Handsontable.Dom.addEvent(button, 'click', function(event) {
+    var changeTypeMenu;
+    var position;
+    var removeMenu;
+
+    document.body.appendChild(menu);
+
+    event.preventDefault();
+    event.stopImmediatePropagation();
+
+    changeTypeMenu = document.querySelectorAll('.changeTypeMenu');
+
+    for (var i = 0, len = changeTypeMenu.length; i < len; i++) {
+      changeTypeMenu[i].style.display = 'none';
+    }
+    menu.style.display = 'block';
+    position = button.getBoundingClientRect();
+
+    menu.style.top = (position.top + (window.scrollY || window.pageYOffset)) + 2 + 'px';
+    menu.style.left = (position.left) + 'px';
+
+    removeMenu = function(event) {
+      if (menu.parentNode) {
+        menu.parentNode.removeChild(menu);
+      }
+    };
+    Handsontable.Dom.removeEvent(document, 'click', removeMenu);
+    Handsontable.Dom.addEvent(document, 'click', removeMenu);
+  });
+}
+
+function _buildDropDownMenu(activeCellType) {
+  var menu = document.createElement('UL');
+  var types = ['text', 'numeric', 'date'];
+  var item;
+
+  menu.className = 'changeTypeMenu';
+
+  for (var i = 0, len = types.length; i < len; i++) {
+    item = document.createElement('LI');
+    if ('innerText' in item) {
+      item.innerText = types[i];
+    } else {
+      item.textContent = types[i];
+    }
+
+    item.data = {'colType': types[i]};
+
+    if (activeCellType === types[i]) {
+      item.className = 'active';
+    }
+    menu.appendChild(item);
+  }
+
+  return menu;
+}
+
+function _buildTypeSwitchButton() {
+  var button = document.createElement('BUTTON');
+
+  button.innerHTML = '\u25BC';
+  button.className = 'changeType';
+
+  return button;
+}
+
+function _isNumeric(value) {
+  if (!isNaN(value)) {
+    if (value.length !== 0) {
+      if (Number(value) <= Number.MAX_SAFE_INTEGER && Number(value) >= Number.MIN_SAFE_INTEGER) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+function _cellRenderer(instance, td, row, col, prop, value, cellProperties, colType) {
+  if (colType === 'numeric' && _isNumeric(value)) {
+    cellProperties.format = '0,0.[00000]';
+    td.style.textAlign = 'left';
+    Handsontable.renderers.NumericRenderer.apply(this, arguments);
+  } else if (value.length > '%html'.length && '%html ' === value.substring(0, '%html '.length)) {
+    td.innerHTML = value.substring('%html'.length);
+  } else {
+    Handsontable.renderers.TextRenderer.apply(this, arguments);
+  }
+}
+
+function _dateValidator(value, callback) {
+  var d = moment(value);
+  return callback(d.isValid() ? true : false);
```

Comment:
do you mind changing this to `return callback(d.isValid());`?

`d.isValid()` already returns boolean, so the ternary creates unnecessary cognitive strain for the reader